include/nuttx/lib/lib.h

```diff
@@ -113,12 +113,11 @@ void lib_stream_release(FAR struct task_group_s *group);
 
 #ifdef CONFIG_STDIO_DISABLE_BUFFERING
 #  define lib_sem_initialize(s)
-#  define lib_take_semaphore(s)
-#  define lib_give_semaphore(s)
+#  define lib_take_semaphore(s, f)
 #else
 void lib_sem_initialize(FAR struct file_struct *stream);
-void lib_take_semaphore(FAR struct file_struct *stream);
-void lib_give_semaphore(FAR struct file_struct *stream);
+int lib_take_semaphore(FAR struct file_struct *stream,
```

Comment:
```suggestion
int  lib_take_semaphore(FAR struct file_struct *stream,
```

Reply:
Thanks, I will add it in a fixup commit.